core/settings.py

```diff
@@ -216,20 +220,193 @@ def as_dict(self):
         """
         d = {}
         for k, v in self.items():
-            if not isinstance(v, Settings):
-                d[k] = v
-            else:
+            if isinstance(v, Settings):
                 d[k] = v.as_dict()
+            elif isinstance(v, list):
+                d[k] = [i.as_dict() if isinstance(i, Settings) else i for i in v]
+            else:
+                d[k] = v
 
         return d
 
 
+    @classmethod
+    def enable_missing(cls):
+        """A context manager for temporary disabling the :meth:`.Settings.__missing__` magic method: all calls now raising a :exc:`KeyError`.
+
+        As a results, attempting to access keys absent from an arbitrary |Settings| instance will raise a :exc:`KeyError`, thus reverting to the default dictionary behaviour.
+
+        .. note::
+            The :meth:`.Settings.__missing__` method is (temporary) deleted at the class level to ensure consistent invokation by the Python interpreter.
+            See also `special method lookup`_.
+
+        Example:
+
+        .. code:: python
+
+            >>> s = Settings()
+
+            >>> with s.enable_missing():
+            ...     s.a.b.c = True
+            KeyError: 'a'
+
+            >>> s.a.b.c = True
+            >>> print(s.a.b.c)
+            True
+
+        .. _`special method lookup`: https://docs.python.org/3/reference/datamodel.html#special-method-lookup
+
+        """
+        return EnableMissing(cls)
+
+
+    def get_nested(self, key_tuple, ignore_missing=True):
+        """Retrieve a nested value by, recursively, iterating through this instance using the keys in *key_tuple*.
+
+        The :meth:`.Settings.__getitem__` method is called recursively on this instance until all keys in key_tuple are exhausted.
+
+        Setting *ignore_missing* to ``False`` will internally open the :meth:`.Settings.enable_missing` context manager, thus raising a :exc:`KeyError` if a key in *key_tuple* is absent from tihs instance.
+
+        .. code:: python
+
+            >>> s = Settings()
+            >>> s.a.b.c = True
+            >>> value = s.get_nested(('a', 'b', 'c'))
+            >>> print(value)
+            True
+        """
+        s = self
+        if ignore_missing:
+            for k in key_tuple:
+                s = s[k]
+        else:  # Ignore Settings.__missing__ and raise a KeyError if a key is missing
+            with s.enable_missing():
```

Comment:
To avoid code duplication here and in `set_nested` it would be better to do something like:
```
with contextlib.supress() if ignore_missing else s.enable_missing():
            for k in key_tuple:
                    s = s[k]
```
See more: https://stackoverflow.com/questions/45187286/how-do-i-write-a-null-no-op-contextmanager-in-python